### PR TITLE
DAOS-9794 build: Build with a generic EL8

### PR DESCRIPTION
**Pull request by brianjmurrell**
Rather than specifying CentOS 8.  This will initially be Rocky Linux
8.5, but could change at any time to be another EL 8 distro such as RHEL
or Almalinux.

### CODE REVIEW

File: utils/rpms/packaging/Dockerfile.centos.7 [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #
-# Copyright 2018-2022, Intel Corporation
+# Copyright 2018-2021, Intel Corporation
```
Comment (brianjmurrell) [inline utils/rpms/packaging/Dockerfile.centos.7:2]: This is reverting an previously unnecessary copyright update for a file that was not actually modified in 2022.  Per discussion, we should only be modifying the copyright on files when we actually modify them.

**Comment by brianjmurrell** [comment]
This PR as is is not going to pass as it requires pipeline-lib's master to be updated to daos-stack/pipeline-lib#287.

A full/successful run with the above PR was run in https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-8206/102/pipeline and
https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-8206/103/pipeline/.

This PR and the above pipeline-lib PR will need to be landed near simultaneously and a rebase will be needed by all PRs once this is landed.  Given that it's Monday morning and the queue is pretty small, this is a perfect time for all of this.

**Review by sylviachanoiyee — APPROVED** [review]

File: docs/QSG/docker.md [hunk 2]
```diff
@@ -95,7 +94,7 @@ This Docker file accept the following arguments:
 	value such as the date of the day shall be given.
 - `DAOS_AUTH`: Enable DAOS authentication when set to "yes" (default "yes")
 - `DAOS_REPOS`: Space separated list of repos needed to install DAOS (default
-	"https://packages.daos.io/v2.0/CentOS8/packages/x86_64/")
+	"https://packages.daos.io/v2.0/EL8/packages/x86_64/")
```
Comment (ashleypittman) [inline docs/QSG/docker.md:97]: This URL isn't valid, is this change intentional?

File: ci/unit/test_main_node.sh [hunk 3]
```diff
@@ -30,8 +30,9 @@ fi
 sudo chown root /usr/bin/daos_admin
 sudo chmod 4755 /usr/bin/daos_admin
 /bin/rm "$DAOS_BASE/install/bin/daos_admin"
-sudo ln -sf "$SL_PREFIX/share/spdk/scripts/setup.sh" /usr/share/spdk/scripts
-sudo ln -sf "$SL_PREFIX/share/spdk/scripts/common.sh" /usr/share/spdk/scripts
+sudo mkdir -p /usr/share/spdk/scripts/
+sudo ln -sf "$SL_PREFIX/share/spdk/scripts/setup.sh" /usr/share/spdk/scripts/
+sudo ln -sf "$SL_PREFIX/share/spdk/scripts/common.sh" /usr/share/spdk/scripts/
```
Comment (ashleypittman) [inline ci/unit/test_main_node.sh:35]: What's changed here other than the extra mkdir call?

File: site_scons/env_modules.py [hunk 4]
```diff
@@ -168,9 +168,9 @@ def show_avail(self):
             print("Could not invoke module avail")
         return output
 
-    def get_map(self):
+    def get_map(self, key):
```
Comment (ashleypittman) [inline site_scons/env_modules.py:171]: Not sure this is related to the PR but looks a good change.

File: src/tests/ftest/harness/basic.py [hunk 5]
```diff
@@ -46,7 +46,7 @@ def test_load_mpi(self):
             self.fail("Orterun initialization failed: {}".format(error))
 
         try:
-            Mpirun(None, mpitype="mpich")
+            Mpirun(None, mpi_type="mpich")
```
Comment (ashleypittman) [inline src/tests/ftest/harness/basic.py:49]: This seems odd, is it related or required?

File: src/tests/ftest/launch.py [hunk 6]
```diff
@@ -1936,7 +1936,7 @@ def install_debuginfos():
 
     # Now install a few pkgs that debuginfo-install wouldn't
     cmd = ["sudo", "dnf", "-y"]
-    if is_el(distro_info):
+    if is_el(distro_info) or "suse" in distro_info.name.lower():
```
Comment (ashleypittman) [inline src/tests/ftest/launch.py:1939]: There would be  case for a "is_suse()" function here.

File: src/tests/ftest/util/exception_utils.py [hunk 7]
```diff
@@ -1,10 +1,11 @@
 #!/usr/bin/python
 """
-  (C) Copyright 2020-2022 Intel Corporation.
+  (C) Copyright 2022 Intel Corporation.
```
Comment (ashleypittman) [inline src/tests/ftest/util/exception_utils.py:3]: This date change appears incorrect.
```suggestion
  (C) Copyright 2020-2022 Intel Corporation.
```

**Review by ashleypittman — COMMENTED** [review]
Minor comments only.

File: src/tests/ftest/util/exception_utils.py (hunk 7 above)
Comment (brianjmurrell) [reply src/tests/ftest/util/exception_utils.py:3]: It's not actually.  It was incorrect before but this corrects it.  This file was created in 2022.

File: src/tests/ftest/launch.py (hunk 6 above)
Comment (brianjmurrell) [reply src/tests/ftest/launch.py:1939]: Perhaps, if we actually had any real SLES to test with.

File: src/tests/ftest/harness/basic.py (hunk 5 above)
Comment (brianjmurrell) [reply src/tests/ftest/harness/basic.py:49]: It's a bug-fix: https://github.com/daos-stack/daos/blob/457d565b632be2b663f7f1316cb2f95051dbb77a/src/tests/ftest/util/job_manager_utils.py#L373-L376

File: site_scons/env_modules.py (hunk 4 above)
Comment (brianjmurrell) [reply site_scons/env_modules.py:171]: It's a bug-fix.

File: ci/unit/test_main_node.sh (hunk 3 above)
Comment (brianjmurrell) [reply ci/unit/test_main_node.sh:35]: Just trailing `/` to make it explicit that the link expects the destination to be a pre-existing directory and not silently creating a `scripts` file.  Belt and suspenders.

**Review by JohnMalmberg — APPROVED** [review]
